### PR TITLE
Determine lib version at compile time

### PR DESCRIPTION
**Pull request by musm**
in this pr:
resolve `libversion` not during  initialization

### CODE REVIEW

**Comment by musm** [comment]
@Allardvm  do you see any problems with this and what are your thoughts?

I was not a fan of the current mechanism of loading the library version name during initialization, which seems very unnecessary and adds to the library load time

File: src/libmagickwand.jl [hunk 1]
```diff
@@ -25,6 +25,15 @@ else
     error("ImageMagick not properly installed. Please run Pkg.build(\"ImageMagick\") then restart Julia.")
 end
 
+function magick_get_libversion(libwand)
```
Comment (musm) [inline src/libmagickwand.jl:28]: this is duplicated in the depfile but I see no way to avoid this...

**Comment by musm** [comment]
ah this is causing precompilation failures on osx for some reason

**Comment by Allardvm** [comment]
Validating that we're loading version 6 makes sense. Let's check 6.0 <= version < 7.0 on all platforms. The crash is because the ENV variables aren't set when you validate. This should be part of the build-time validation function (just copy paste the ENV setup for each platform from the preload stuff).

**Comment by musm** [comment]
I have removed the validation check and restricted this PR to just  get the library version at compile time instead of  at runtime

**Comment by timholy** [comment]
More OSX failures...

**Comment by musm** [comment]
closed since no longer relevant

**Pull request closed without merging**